azurerm/internal/services/mssql/validate/mssql_database.go

```diff
@@ -0,0 +1,75 @@
+package validate
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/mssql/parse"
+)
+
+func MsSqlDatabaseID(i interface{}, k string) (warnings []string, errors []error) {
+	v, ok := i.(string)
+	if !ok {
+		errors = append(errors, fmt.Errorf("expected type of %q to be string", k))
+		return warnings, errors
+	}
+
+	if _, err := parse.MsSqlDatabaseID(v); err != nil {
+		errors = append(errors, fmt.Errorf("Can not parse %q as a MsSql Database resource id: %v", k, err))
+	}
+
+	return warnings, errors
+}
+
+func MsSqlDatabaseAutoPauseDelay(i interface{}, k string) (warnings []string, errors []error) {
+	v, ok := i.(int)
+	if !ok {
+		errors = append(errors, fmt.Errorf("expected type of %s to be integer", k))
+		return warnings, errors
+	}
+	min := 60
+	max := 10080
+	if (v < min || v > max) && v%10 != 0 && v != -1 {
+		errors = append(errors, fmt.Errorf("expected %s to be in the range (%d - %d) and divisible by 10 or -1, got %d", k, min, max, v))
```

Comment:
This shouldn't be able to be `-1`?
```suggestion
		errors = append(errors, fmt.Errorf("expected %s to be in the range (%d - %d) and divisible by 10, got %d", k, min, max, v))
```

Reply:
In the API, they suppose `-1` as disable `auto pause delay`. So I explain it in the resource doc:`A value of -1 means that automatic pause is disabled`.